src/builtins/complete.cpp

```diff
@@ -394,6 +394,10 @@ maybe_t<int> builtin_complete(parser_t &parser, io_streams_t &streams, const wch
             completion_list_t comp = complete(
                 do_complete_param, completion_request_options_t::normal(), parser.context());
 
+            if (have_do_complete_param) {
```

Comment:
"have_do_complete_param" means that we're not using the current commandline.

It's this:

```fish
complete -C"git checkout "
# vs
complete -C # <- this uses the current commandline, the argument is optional!
```

Notably it doesn't apply to `complete` (which lists all defined completions - `complete -c foo -l foo-option1` and so on).

If we are in this branch, we already know we are in "do_complete" mode, i.e. we are running completions to return completion candidates.

I'm not entirely sure why we wouldn't run the same logic on the current commandline?

Reply:
Oh, I see. In that case, I should revisit it and remove that precondition.